src/node_js/closures.rs

```diff
@@ -252,6 +355,29 @@ macro_rules! impls {(
     >
         Closure_<($($_0, $($_k ,)*)?), CRet>
     {
+        /// Make Node.js's main event loop wait for the `Closure` to be
+        /// dropped before ending.
+        ///
+        /// If used incorrectly, this can make Node.js keep spinning infinitely
+        /// at the end of the program execution.
+        ///
+        /// On the other hand, if you have a `Closure` that _needs_ to be
+        /// called (or dropped), and which you know _will be **dropped**_ before
+        /// Node.js reaches the end of its main loop, then you can call this
+        /// function with `true`.
+        pub
+        fn make_nodejs_wait_for_this_to_be_dropped (
+            self: &'_ mut Self,
+            nodejs_should_wait: bool,
+        ) -> Result<()>
+        {
+            if nodejs_should_wait {
+                self.ts_fun.refer(&self.env)
+            } else {
+                self.ts_fun.unref(&self.env)
```

Comment:
If we already `unref` by default then does this need to be here?

Reply:
I can envision somebody hitting the use case of having called `.nodejs_should_wait(true);` given a situation where they know node.js needs to wait for that closure for a while, but for some reason later on (I imagine based off some dynamic condition), it no longer being the case. They'd then need a way to "undo" that `(true)` call, which would be done by doing `.nodejs_should_wait(false); /* not anymore */`.
Does that seem like a legitimate use case? I may be doing speculative work 😅 , but it's just a boolean branch so I think it's okay 🙂

Reply:
Yeah, that makes sense. 👍